src/hooks/use-feature-flags.tsx

```diff
@@ -26,15 +26,24 @@ interface FeatureFlagsProviderProps {
 }
 
 export const FeatureFlagsProvider: React.FC< FeatureFlagsProviderProps > = ( { children } ) => {
-	const [ featureFlags, setFeatureFlags ] = useState< FeatureFlagsContextType >( {
-		...defaultFeatureFlags,
+	const [ featureFlags, setFeatureFlags ] = useState< FeatureFlagsContextType >( () => {
+		if ( window.appGlobals ) {
+			return {
+				...defaultFeatureFlags,
+				...window.appGlobals,
+			};
+		}
+		return { ...defaultFeatureFlags };
```

Comment:
Hm, can't we go this way? it should work:
<img width="253" height="91" alt="Screenshot 2025-07-31 at 18 53 48" src="https://github.com/user-attachments/assets/aa252c17-0284-4a2b-ae5b-3371b7e815cc" />

```suggestion
		return {
			...defaultFeatureFlags,
			...window.appGlobals,
		};
```